table/tables/partition_test.go

```diff
@@ -447,6 +447,25 @@ func (ts *testSuite) TestCreatePartitionTableNotSupport(c *C) {
 	c.Assert(ddl.ErrPartitionFunctionIsNotAllowed.Equal(err), IsTrue)
 }
 
+// issue 24880
+func (ts *testSuite) TestRangePartitionUnderNoUnsignedSub(c *C) {
+	tk := testkit.NewTestKitWithInit(c, ts.store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists tu;")
+	tk.MustExec("SET @@sql_mode='NO_UNSIGNED_SUBTRACTION';")
+	_, err := tk.Exec(`CREATE TABLE tu (c1 BIGINT UNSIGNED) PARTITION BY RANGE(c1 - 10) (
+						PARTITION p0 VALUES LESS THAN (-5),
+						PARTITION p1 VALUES LESS THAN (0),
+						PARTITION p2 VALUES LESS THAN (5),
+						PARTITION p3 VALUES LESS THAN (10),
+						PARTITION p4 VALUES LESS THAN (MAXVALUE)
+						);`)
+	c.Assert(err, IsNil)
+	// currently not support insert records whose partition value is negative
+	_, err = tk.Exec(("insert into tu values (0);"))
```

Comment:
can you please also insert a big unsigned value?

Reply:
Update in the new commit.

Reply:
a value greater than `bigint signed` but less than `bigint unsigned` is expected.

Reply:
Update in the new commit.